### DISCUSSION TITLE
problem with subdomain redirect

### DISCUSSION THREAD

**Discussion by ghalandar** [community/community#75861] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

hello,
I use test.com for one repository and sub.test.com for another repository . Both of them are connected to the address of their repositories with a CNAME record in the DNS panel.
for example:
www > repo1.github.io (test.com)
sub > repo2.github.io (sub.test.com)
If we go to repo1.github.io, it will redirect to test.com.
But the problem is that if we go to repo2.github.io, it does not redirect to sub.test.com and gives a 404 error.
how can i fix it?

**Comment by queenofcorgis** [comment]
Are you able to share the repositories? Sharing them may help better diagnose and identify the problem.

**Comment by ghalandar** [comment, marked as answer]
So with the `*.github.io` `CNAME`records, you'd actually want to point those to the user/org name, and not the repository. For example: `test.github.io` should be the content for the `CNAME`'s if they're hosted on your test org. If they're hosted on your account, you'd want to change it to `youraccount.github.io`. For more info on that, you can see it from [this article](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#dns-records-for-your-custom-domain).

>  it does not redirect to sub.test.com and gives a 404 error.

This is due to only your username can be rerouted as a `*.github.io` username.

**Comment by queenofcorgis** [comment]
We’re glad you could get this solved and we want to make sure others can do the same so we’ve reopened this discussion for better discoverability for our members.
